src/dragonball/src/device_manager/balloon_dev_mgr.rs

```diff
@@ -0,0 +1,435 @@
+// Copyright 2020 Alibaba Cloud. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use dbs_virtio_devices as virtio;
+use serde_derive::{Deserialize, Serialize};
+use slog::{error, info};
+use virtio::balloon::{Balloon, BalloonConfig};
+use virtio::Error as VirtIoError;
+
+use crate::address_space_manager::GuestAddressSpaceImpl;
+use crate::config_manager::{ConfigItem, DeviceConfigInfo, DeviceConfigInfos};
+use crate::device_manager::DbsMmioV2Device;
+use crate::device_manager::{DeviceManager, DeviceMgrError, DeviceOpContext};
+
+// The flag of whether to use the shared irq.
+const USE_SHARED_IRQ: bool = true;
+// The flag of whether to use the generic irq.
+const USE_GENERIC_IRQ: bool = false;
+
+/// Errors associated with `BalloonDeviceConfig`.
+#[derive(Debug, thiserror::Error)]
+pub enum BalloonDeviceError {
+    /// The balloon device was already used.
+    #[error("the virtio-balloon sock path was already added to a different device")]
```

Comment:
Why this is `sock path`?
I read through the code, mostly is `id` conflicts but I don't see any `sock path`

Reply:
you're right! this is fixed